conda-store-server/conda_store_server/action/generate_constructor_installer.py

```diff
@@ -53,11 +56,38 @@ def write_file(filename, s):
     # conda and pip need to be in dependencies for the post_install script
     dependencies = ["conda", "pip"]
     pip_dependencies = []
-    for d in specification.dependencies:
-        if type(d) is schema.CondaSpecificationPip:
-            pip_dependencies.extend(d.pip)
-        else:
-            dependencies.append(d)
+    channels = []
+
+    if is_lockfile:
+        # Adds channels
+        channels = [c.url for c in specification.lockfile.metadata.channels]
+
+        # Adds dependencies
+        for p in specification.lockfile.package:
+            # Ignores packages not matching the current platform. Versions can
+            # be different between platforms or a package might not support all
+            # platforms. constructor is cross-friendly, but we're currently
+            # building only for the current architecture, see the comment in
+            # get_installer_platform
+            if p.platform != get_installer_platform():
```

Comment:
Is this excluding `noarch` too? It shouldn't.

Reply:
Updated. However, it might be unnecessary here. I've tried installing two `noarch` packages. They still have the platform set to `linux-64` despite coming from a `noarch` url. I think it's because of how we call `action_solve_lockfile`, which always passes `conda_utils.conda_platform()` to conda-lock.

<details>
<summary>Repro</summary>

spec:
```
channels:
- conda-forge
dependencies:
- pytest
- pip:
  - flask
- conda-forge/noarch::pytest-timeout
- conda-forge/noarch::pytest-mock
description: ''
name: test
prefix: null
variables: null
```

lockfile:
```
        {
            "name": "pytest-mock",
            "version": "3.14.0",
            "manager": "conda",
            "platform": "linux-64",
            "dependencies": {
                "python": ">=3.8",
                "pytest": ">=5.0"
            },
            "url": "https://conda.anaconda.org/conda-forge/noarch/pytest-mock-3.14.0-pyhd8ed1ab_0.conda",
            "hash": {
                "md5": "4b9b5e086812283c052a9105ab1e254e",
                "sha256": "f7cd8910ce3690a9189483f4b01dde1970b2b86ea853a03b1225f344f988045d"
            },
            "category": "main",
            "optional": false
        },
        {
            "name": "pytest-timeout",
            "version": "2.3.1",
            "manager": "conda",
            "platform": "linux-64",
            "dependencies": {
                "python": ">=3.7",
                "pytest": ">=7.0.0"
            },
            "url": "https://conda.anaconda.org/conda-forge/noarch/pytest-timeout-2.3.1-pyhd8ed1ab_1.conda",
            "hash": {
                "md5": "27860d8e6cc9e14da4ea900788693e40",
                "sha256": "501e2417f23596815aeb41ad3bc69016a59c9f5dab62860b8be41e006d538fe8"
            },
            "category": "main",
            "optional": false
        },
```

</details>